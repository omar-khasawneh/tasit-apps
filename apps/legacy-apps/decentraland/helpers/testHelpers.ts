import { ESTATE, PARCEL } from "../constants/AssetTypes";
import { DONE, PENDING, MISSING } from "../constants/ActionStatus";
import { SUCCESSFUL } from "../constants/UserActionStatus";

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateWithoutName = {
  type: ESTATE,
  id: "321",
  name: "",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateForSale = {
  id: "1",
  priceMana: "0",
  priceUSD: 0,
  asset: estate,
};

export const parcelForSale = {
  id: "2",
  priceMana: "0",
  priceUSD: 0,
  asset: parcel,
};

function getIdAsync(): Promise<string> {
  return new Promise((resolve, reject) => {
    process.nextTick(() =>
      resolve("0x1234567890123456789012345678901234567890")
    );
  });
}

export const anAction = {
  getId: async (): Promise<string> => {
    return await getIdAsync();
  },
};

export const accountCreationSteps = [
  {
    name: "Account created",
    status: DONE,
  },
  {
    name: "ETH",
    status: PENDING,
  },
  {
    name: "MANA",
    status: MISSING,
  },
];

export const parcelUserAction = {
  ["0x1234567890123456789012345678901234567890"]: {
    assetId: "0123456789",
    status: SUCCESSFUL,
  },
};

export const estateUserAction = {
  ["0x987654321098765432109876543210987654321"]: {
    assetId: "123",
    status: SUCCESSFUL,
  },
};

export const userActions = { ...parcelUserAction, ...estateUserAction };

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
  anAction,
  estateWithoutName,
  parcelUserAction,
  estateUserAction,
  userActions,
};
