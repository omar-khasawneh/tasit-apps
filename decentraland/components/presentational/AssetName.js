import React from "react";
import { StyleSheet, View, Text } from "react-native";
import PropTypes from "prop-types";
import { responsiveHeight } from "react-native-responsive-dimensions";
import Colors from "@constants/Colors";

export function AssetName({ asset }) {
  let { name } = asset;
  if (!name) name = "(No name for this one right now)";

  return (
    <View style={styles.landNameContainer}>
      <Text style={styles.landName}>{name}</Text>
    </View>
  );
}

AssetName.propTypes = {
  asset: PropTypes.object.isRequired,
};

const styles = StyleSheet.create({
  landNameContainer: {
    paddingTop: responsiveHeight(1),
  },
  landName: {
    color: Colors.assetInfoText,
    fontWeight: "bold",
  },
});

export default AssetName;
