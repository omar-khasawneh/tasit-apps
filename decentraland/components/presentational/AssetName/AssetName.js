import React from "react";
import { StyleSheet, View, Text } from "react-native";

import { responsiveHeight } from "react-native-responsive-dimensions";
import Colors from "@constants/Colors";

export function AssetName({ name }) {
  if (!name) name = "(No name for this one right now)";

  return (
    <View style={styles.landNameContainer}>
      <Text style={styles.landName}>{name}</Text>
    </View>
  );
}

// TODO: Migrate me to TypeScript types
AssetName.propTypes = {
  name: PropTypes.string.isRequired,
};

const styles = StyleSheet.create({
  landName: {
    color: Colors.assetInfoText,
    fontWeight: "bold",
  },
  landNameContainer: {
    paddingTop: responsiveHeight(1),
  },
});

export default AssetName;
