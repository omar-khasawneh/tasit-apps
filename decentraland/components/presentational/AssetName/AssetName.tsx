import React from "react";
import { StyleSheet, View, Text } from "react-native";
import { responsiveHeight } from "react-native-responsive-dimensions";
import Colors from "../../../constants/Colors";
type AssetNameProps = {
  name: string;
};
export const AssetName: React.SFC<AssetNameProps> = ({ name }) => {
  if (!name) name = "(No name for this one right now)";
  return (
    <View style={styles.landNameContainer}>
      <Text style={styles.landName}>{name}</Text>
    </View>
  );
};
const styles = StyleSheet.create({
  landName: {
    color: Colors.assetInfoText,
    fontWeight: "bold",
  },
  landNameContainer: {
    paddingTop: responsiveHeight(1),
  },
});
export default AssetName;
