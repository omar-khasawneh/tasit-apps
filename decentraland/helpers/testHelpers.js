import AssetTypes from "@constants/AssetTypes";
import { DONE, PENDING, MISSING } from "@constants/ActionStatus";
const { ESTATE, PARCEL } = AssetTypes;

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateWithoutName = {
  type: ESTATE,
  id: "321",
  name: "",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateForSale = {
  id: "1",
  priceMana: "0",
  priceUSD: 0,
  asset: estate,
};

export const parcelForSale = {
  id: "2",
  priceMana: "0",
  priceUSD: 0,
  asset: parcel,
};

export const anAction = {
  getId: async () => "0x1234567890123456789012345678901234567890",
};

export const accountCreationSteps = [
  {
    name: "Account created",
    status: DONE,
  },
  {
    name: "ETH",
    status: PENDING,
  },
  {
    name: "MANA",
    status: MISSING,
  },
];

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
  anAction,
  estateWithoutName,
  accountCreationSteps,
};
