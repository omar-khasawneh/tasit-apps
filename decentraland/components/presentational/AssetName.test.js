import React from "react";
import { shallow } from "enzyme";
import AssetName from "./AssetName";
import { parcel, estateWithoutName } from "@helpers/testHelpers";

describe("AssetName", () => {
  describe("renders the component", () => {
    it("asset with name", async () => {
      const asset = parcel;
      expect(shallow(<AssetName asset={asset} />)).toMatchSnapshot();
    });

    it("asset without name", async () => {
      const asset = estateWithoutName;
      expect(shallow(<AssetName asset={asset} />)).toMatchSnapshot();
    });
  });
});
