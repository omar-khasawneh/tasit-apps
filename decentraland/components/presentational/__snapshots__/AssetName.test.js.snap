// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetName renders the component asset with name 1`] = `
<Component
  style={
    Object {
      "paddingTop": 13.34,
      "width": 375,
    }
  }
>
  <Component
    style={
      Object {
        "color": "#2B2B2B",
        "fontWeight": "bold",
      }
    }
  >
    Sample Parcel
  </Component>
</Component>
`;

exports[`AssetName renders the component asset without name 1`] = `
<Component
  style={
    Object {
      "paddingTop": 13.34,
      "width": 375,
    }
  }
>
  <Component
    style={
      Object {
        "color": "#2B2B2B",
        "fontWeight": "bold",
      }
    }
  >
    (No name for this one right now)
  </Component>
</Component>
`;
